test/apiserver_organization_test.go

```diff
@@ -135,22 +117,212 @@ modfor:
+}
+
+func TestVisibleFiltering(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	cfg, err := config.GetConfig()
+	require.NoError(t, err)
+	cfg.UserAgent = t.Name()
+	cl := testutil.NewRecordingClient(t, cfg, testScheme, testutil.CleanUpStrategy(cleanUpStrategy))
+	t.Cleanup(cl.CleanUpFunc(ctx))
+	dcl, err := dynamic.NewForConfig(cfg)
+	require.NoError(t, err)
+	wi, err := dcl.Resource(gvr).Watch(ctx, metav1.ListOptions{
+		LabelSelector: "test-name=" + t.Name(),
+	})
+	require.NoError(t, err)
+	globalEventTraced := events.NewTracer(wi)
+	t.Cleanup(globalEventTraced.TestCleanupFunc(t))
+
+	owner := rbacv1.Subject{
+		Kind:     "User",
+		APIGroup: "rbac.authorization.k8s.io",
+		Name:     "kubernetes-admin",
+	}
+	testCase := []*TestVisibleFilteringTestCase{
+		{
+			Name: "user",
+			Subject: rbacv1.Subject{
+				Kind:     rbacv1.UserKind,
+				APIGroup: rbacv1.GroupName,
+				Name:     "user",
+			},
+			ImpersonationConfig: rest.ImpersonationConfig{
+				UserName: "user",
+			},
+		},
+		{
+			Name: "group",
+			Subject: rbacv1.Subject{
+				Kind:     rbacv1.GroupKind,
+				APIGroup: rbacv1.GroupName,
+				Name:     "group",
+			},
+			ImpersonationConfig: rest.ImpersonationConfig{
+				UserName: "lala",
+				// without "system:authenticated" things are breaking...cannot RESTMapper doesn't function
+				Groups: []string{"system:authenticated", "group"},
+			},
+		},
+		{
+			Name: "sa",
+			Subject: rbacv1.Subject{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      "default",
+				Namespace: "default",
+			},
+			ImpersonationConfig: rest.ImpersonationConfig{
+				UserName: "system:serviceaccount:default:default",
+			},
+		},
+	}
+	t.Log("ensuring cluster RBAC")
+	ensureClusterRBAC(t, ctx, cl, testCase)
+
+	t.Log("creating orgs")
+	for _, tc := range testCase {
+		cfg, err := ctrl.GetConfig()
+		require.NoError(t, err)
+		cfg.Impersonate = tc.ImpersonationConfig
+		cfg.UserAgent = t.Name() + "/" + tc.Name
+		tc.Client = testutil.NewRecordingClient(t, cfg, testScheme, testutil.CleanUpStrategy(cleanUpStrategy))
+		t.Cleanup(tc.Client.CleanUpFunc(ctx))
+		dcl, err := dynamic.NewForConfig(cfg)
+		require.NoError(t, err)
+		wi, err := dcl.Resource(gvr).Watch(ctx, metav1.ListOptions{
+			LabelSelector: "test-name=" + t.Name(),
+		})
+		require.NoError(t, err)
+		tc.tracer = events.NewTracer(wi, events.IsObjectName("test-"+tc.Name))
+		t.Cleanup(tc.tracer.TestCleanupFunc(t))
+		tc.Org = &apiserverv1alpha1.Organization{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test-" + tc.Name,
+				Labels: map[string]string{
+					"test-name": t.Name(),
+				},
+			},
+			Spec: storagev1alpha1.OrganizationSpec{
+				Metadata: &storagev1alpha1.OrganizationMetadata{
+					DisplayName: "test",
+					Description: "desc",
+				},
+				Owners: []rbacv1.Subject{owner},
+			},
+		}
+		require.Error(t, tc.Client.Create(ctx, tc.Org), "I cannot create an organization which I'm not owner of")
```

Comment:
This error message is kind of misleading... Owner doesn't have permission to do anything with Organization, no matter of if he/she is the owner of the Organization.

Reply:
Hmm, I think we should talk about the permission for managing Organization object during our daily...because from concept doc, we don't predefine OrganizationRoleTempalte for Owner to manage Organization object, but now the more I think about it, I think ower should have permission to that.
WDYT? I can add another OrganizationRoleTemplate to handle that after this PR gets merged, not a big deal.

Reply:
we can talk during the daily about details, for now I've updated the failed message description

Reply:
thanks!